lib/europeana/proxy/media.rb

```diff
@@ -96,7 +97,13 @@ def proxy?(env)
       # @return [Hash] rewritten request env
       def rewrite_env(env)
         env['app.record_id'] = env['REQUEST_PATH']
-        edm = Europeana::API.record(env['app.record_id'])['object']
+
+        if env['app.params']['api_url']
+          fail Errors::AccessDenied, 'Requested API url is invalid' unless @permitted_api_urls.include?(env['app.params']['api_url'])
+          Europeana::API.url = env['app.params']['api_url']
```

Comment:
We should not be setting this at the `Europeana::API` module level as it would affect future requests having no `api_url` param. Instead, `api_url` should just be passed in the call to `Europeana::API.record.fetch`.